apps/greencheck/microsoft_bucket_updater.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import requests
+
+from datetime import datetime, timedelta
+from django.conf import settings
+from . import object_storage
+
+
+logger = logging.getLogger(__name__)
+
+
+class MicrosoftBucketUpdater:
+    def __init__(self):
+        self.url_prefix = "https://download.microsoft.com/download/7/1/D/71D86715-5596-4529-9B13-DA13A5DE5B63/ServiceTags_Public_"
+        self.url_file_extension = ".json"
+
+    def search_dataset(self) -> dict:
+        """
+        Searches for a working url from Microsoft's endpoint.
+        If a dataset is found, it will be returned by this function.
+        Return: dict
+        """
+        date_to_check = datetime.now()
+        date_threshold = datetime.now() - timedelta(days=5)
+
+        # Iterate through the days to find a working
+        # endpoint that returns a dataset
+        while date_to_check >= date_threshold:
+            response = requests.get(self.format_date_to_url(date_to_check))
+
+            if response.status_code == 200:
```

Comment:
See above. returning just the URL would mean this method is only telling us if a url is good to use, not returning any content. We already have methods for fetching the code,  and personally, I'd prefer less complex, more composable code over saving on making HTTP requests.